### PR TITLE
py-markdown: Add new package for python-markdown.

### PR DESCRIPTION
**Pull request by cmacmackin**
Added a package for [Python Markdown](https://pythonhosted.org/Markdown/), the most often used Markdown implementation in Python.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-markdown/package.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    version('2.6.4', '5fb3cd9945eb534e71af597f8ee3622b')
+    version('2.6.3', 'ec7a50ce9fd4a5fd0b24555d47e9d7d1')
+    version('2.6.2', '6ce86913e9bf5bb34d9ee394ac71f044')
+    version('2.6.1', '0ae69693c5adb27caf0160941d7dcbdf')
+    version('2.6', '9acdde43d99847d0c4ef03ea56b1d2c5')
+    version('2.5.2', 'ed2a662d22799186c1ef85d173d38b8a')
+    version('2.5.1', 'be6f6ba65a8fb843d2aaf1fcdd68c755')
+    version('2.5', '8393ceab9c6e33357fb8a7be063a4849')
+
+    extends('python')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-markdown/package.py]: Can you update this with the constraints on Python version just so a user doesn't try to build it with an incompatible version? For the latest version, it looks like:
```python
extends('python')
depends_on('python@2.7:2.8,3.2:3.4')
```

File: var/spack/repos/builtin/packages/py-markdown/package.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    depends_on('py-setuptools', type='build')
+
+    def url_for_version(self, version):
+        base_url = "https://github.com/waylan/Python-Markdown/archive"
+        return "{0}/{1}-final.tar.gz".format(base_url, version)
```
Comment (citibeth) [inline var/spack/repos/builtin/packages/py-markdown/package.py]: Get rid of {0} here, it serves no purpose.
Also... is `url_for_version()` necessary here?  I believe Spack should do the same thing given the `url` sring above.

File: var/spack/repos/builtin/packages/py-markdown/package.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+class PyMarkdown(Package):
+    """This is a Python implementation of John Gruber's Markdown. It is
+    almost completely compliant with the reference implementation, though
+    there are a few very minor differences. See John's Syntax
+    Documentation for the syntax rules.
+    """
+
+    homepage = "https://pythonhosted.org/Markdown/"
+    url      = "https://github.com/waylan/Python-Markdown/archive/"\
```
Comment (citibeth) [inline var/spack/repos/builtin/packages/py-markdown/package.py]: PEP-8 specifies to not line things up with spaces.

File: var/spack/repos/builtin/packages/py-markdown/package.py (hunk 1 above)
Comment (cmacmackin) [inline var/spack/repos/builtin/packages/py-markdown/package.py]: Done.

**Comment by citibeth** [comment]
Sorry to be a pain... but I would suggest we use the following for Python version instead:
```depends_on('python@2.7:2.8,3.2:')```

It might not have been tested on Python 3.5, but it almost certainly works there.  I'd do it that way until/unless we learn that it does not work on Python 3.5 (and above).

**Comment by adamjstewart** [comment]
@citibeth But doesn't setuptools crash when you try to build it with a version of Python that isn't explicitly supported? That's why we had to hack up `py-pyside` to convince it that it can actually build with Python 3.5, right?

**Comment by citibeth** [comment]
Aw c**p, I think you're right.  OK, best to leave it at `:3.4` until
someone tries / gets it working on 3.5.

**Comment by adamjstewart** [comment]
@cmacmackin Can you remove the last commit? After that I think everything should be good to go.

**Comment by cmacmackin** [comment]
Flake8 had a small complaint with this file, so I ended up adding a new commit which fixed that and re-added the upper bound to the Python version.

**Pull request merged**